aten/src/ATen/native/mobile/cpu/internal/Convolution.cpp

```diff
@@ -0,0 +1,349 @@
+      Spatial(Layout::Filter::Width, Layout::Parameter::Width),
+    },
+  };
+}
+
+bool usable(const Tensor& input) {
+         // Input
+  return (4 == input.ndimension()) &&
+         (c10::DeviceType::CPU == input.device().type()) &&
+         (kFloat == input.scalar_type()) &&
+         (input.size(Layout::Activation4D::Batch) > 0) &&
+         (input.size(Layout::Activation4D::Channels) > 0) &&
+         (input.size(Layout::Activation4D::Height) > 0) &&
+         (input.size(Layout::Activation4D::Width) > 0);
+}
+
+Tensor run(
+    const Context& context,
+    const Tensor& input) {
+  using namespace mobile::internal;
+
+  TORCH_INTERNAL_ASSERT(
+      usable(input),
+      "mobile::cpu::convolution not usable!");
+
+  const auto Spatial = [](const int64_t input, const Context::Output::Spatial spatial) {
+    return (input + spatial.alpha) / spatial.beta;
+  };
+
+  Tensor output = new_tensor(
+      std::array<int64_t, 4>{
+        input.size(Layout::Activation4D::Batch),
+        context.output.channels,
+        Spatial(input.size(Layout::Activation4D::Height), context.output.height),
+        Spatial(input.size(Layout::Activation4D::Width), context.output.width),
+      },
+      input.options(),
+      MemoryFormat::ChannelsLast);
+
+  const xnn_status setup_status = xnn_setup_convolution2d_nhwc_f32(
+      context.convolution_op.get(),                                   // operator
+      input.size(Layout::Activation4D::Batch),                        // batch_size
+      input.size(Layout::Activation4D::Height),                       // input_height
+      input.size(Layout::Activation4D::Width),                        // input_width
+      input.contiguous(MemoryFormat::ChannelsLast).data_ptr<float>(), // input
+      output.data_ptr<float>(),                                       // output
+      threadpool().handle());                                         // threadpool
+
+  TORCH_INTERNAL_ASSERT(
+      xnn_status_success == setup_status,
+      "xnn_setup_convolution2d_nhwc_f32 failed!");
+
+  const xnn_status run_status = xnn_run_operator(
+      context.convolution_op.get(), // operator
+      threadpool().handle());       // threadpool
+
+  TORCH_INTERNAL_ASSERT(
+      xnn_status_success == run_status,
+      "xnn_run_operator failed!");
+
+  return output.contiguous(input.suggest_memory_format());
+}
+
+Tensor create_and_run(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const IntArrayRef padding,
+    const IntArrayRef stride,
+    const IntArrayRef dilation,
+    const int64_t groups,
+    const bool transposed,
+    const Scalar output_min,
+    const Scalar output_max) {
+  return run(
+      create(
+          weight,
+          bias,
+          padding,
+          stride,
+          dilation,
+          groups,
+          transposed,
+          output_min,
+          output_max),
+      input);
+}
+
+const auto registry = c10::RegisterOperators()
+  .op("mobile::conv2d_create",
```

Comment:
`native_functions.yaml` doesn't support namespaces like `mobile::` yet, but generally yes, it's usually better to define things in `native_functions.yaml` because you get the C++ frontend generated.